src/main/java/net/minecraftforge/server/permission/nodes/PermissionTypes.java

```diff
@@ -17,19 +17,17 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
-package net.minecraftforge.server.permission;
+package net.minecraftforge.server.permission.nodes;
+
+import net.minecraft.network.chat.Component;
 
 /**
- * <table><thead><tr><th>Level</th><th>Player</th><th>OP</th></tr>
- * </thead><tbody>
- * <tr><td>ALL</td><td>true</td><td>true</td></tr>
- * <tr><td>OP</td><td>false</td><td>true</td></tr>
- * <tr><td>NONE</td><td>false</td><td>false</td></tr>
- * </tbody></table>
+ * Default PermissionTypes, if you need additional ones, please PR it.
  */
-public enum DefaultPermissionLevel
+public class PermissionTypes
 {
-    ALL,
-    OP,
-    NONE
+   public static final PermissionType<Boolean> BOOL = new PermissionType<>(Boolean.class, "Boolean");
+   public static final PermissionType<Integer> INT = new PermissionType<>(Integer.class, "Integer");
+   public static final PermissionType<String> STRING = new PermissionType<>(String.class, "String");
+   public static final PermissionType<Component> COMPONENT = new PermissionType<>(Component.class, "Component");
```

Comment:
Uppercase names? Also, some Javadoc on examples of usage might be neat. It's probably redundant for booleans, but components and strings would benefit IMO.